### DISCUSSION TITLE
Postal keep using Ipv6 for outgoing emails instead of the IPv4

### DISCUSSION THREAD

**Discussion by snipecoder** [postalserver/postal#3010] [Help with using Postal]
Hello

I have set up IP pools and configured my IPS to use them. However, instead of using these IPs, he is using IPv6 and I don't understand why.

My second question is, how can I enable sending from any domain, even if it's not added to the mail server, but still be able to receive bounce notifications for this domain?

Thank you

**Comment by willpower232** [comment]
> However, instead of using these IPs, he is using IPv6

How are you seeing this? Are you able to share a screenshot of what looks wrong?

> how can I enable sending from any domain, even if it's not added to the mail server

I think this is what the sender header setting is for

**Reply by snipecoder** [reply]
Yes, see the following - also I am missing all the headers like List-Unsubscribe or others, should postal add this?

`spf=softfail (google.com: domain of transitioning zidlyk@psrp.send.domain.com does not designate 2607:5300:203:9663:: as permitted sender) `

`Received: from send.domain.com ([2607:5300:203:9663::])`

I have no idea where the postal got this IP from. I only set up IPv4 in the IP Pools and checked to use that IP in the mail server I have created.

**Reply by willpower232** [reply]
You would have to ask your server provider in case they are using some form of 4-6 process if it isn't immediately obvious to you.

**Comment by snipecoder** [comment]
Another question, is there any option to reject Sender/Domain when the bounce rate is above X%?

**Reply by willpower232** [reply]
I don't believe so, Postal is not intended to retain data long term so you would have to implement webhooks and your own blocks within whatever you are sending with.

**Reply by snipecoder** [reply]
Is it possible to use webhook in postal, before send, check the bounce rate in mautic/mailwizz and reject the message?

**Reply by willpower232** [reply]
Postal only sends webhooks when the status of the message changes after it has been sent, there is not a webhook or event or anything for before it is sent as presumably the application generating the message should have checked that before giving the message to Postal.

**Reply by snipecoder** [reply]
Can I create a custom webhook/plugin for this?
I need to use API to check the external system for bounce rate and more info and put conditions based on the response

Which MTA does Postal use? Where can I edit such things?

Is anyone here available for small customizations?

If I set ALLOW SENDER HEADER to true, will it still be tracked when the domain is not added to the domains list?(Bounces/Return-Path/Spam, etc)

**Reply by snipecoder** [reply]
I am continuing to receive this error:
`550-5.7.25 [2607:5300:203:9663::] The IP address sending this message does not 550-5.7.25 have a PTR record setup, or the corresponding forward DNS entry does 550-5.7.25 not match the sending IP. As a policy, Gmail does not accept 550-5.7.25 messages`

But I have configured IP Pools:
![image](https://github.com/postalserver/postal/assets/99405975/1aeedacc-a280-4751-be98-4fcff33d5669)
![image](https://github.com/postalserver/postal/assets/99405975/590d25f0-6dbb-4c56-b5de-ff964b40ed7e)

What is the reason that, despite setting IPv4 in the pool settings, Postal tries to send via IPv6? (IPv4 is working perfectly on the same server, where I have another MTA running.)

**Reply by stef157** [reply]
Hello,

Have you managed to send emails only from IPv4? I'm encountering exactly the same issue as you...

Thanks
